File: src/main/java/br/com/leonardoferreira/jirareport/client/FieldClient.java
package br.com.leonardoferreira.jirareport.client;

import br.com.leonardoferreira.jirareport.domain.vo.JiraField;
import java.util.List;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestHeader;

@FeignClient(name = "field-client", url = "${jira.url}")
public interface FieldClient {

    @GetMapping("/rest/api/2/field")
    List<JiraField> findAll(@RequestHeader("Authorization") String token);

}

File: client-library/src/main/kotlin/br/com/jiratorio/client/FieldClient.kt
package br.com.jiratorio.client

import br.com.jiratorio.JiraClientConfiguration
import br.com.jiratorio.jira.JiraField
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping

@FeignClient(
    name = "field-client",
    url = "\${jira.url}",
    configuration = [
        JiraClientConfiguration::class
    ]
)
interface FieldClient {

    @GetMapping("/rest/api/2/field")
    fun findAll(): List<JiraField>

}
